### PR TITLE
LPD-13338

### PR DESCRIPTION
**Pull request by ealonso**
Motivation
-----
Allow mark a section as deprecated in **Form Navigation Taglib**.

Widget_Page_-_Liferay

[Link to the ticket](https://liferay.atlassian.net/browse/LPD-13311)
[Link to the Feature Request](https://liferay.atlassian.net/browse/LPD-557)

Proposed Solution
-----

Adds the following method to FormNavigatorEntry:

```
public default boolean isDeprecated() {
      return false;
}
```

And adds **deprecated** attribute to FieldsetTag.

In that way we can show or not the deprecated label into the title of a section.

### CODE REVIEW

**Review by markocikos — APPROVED** [review]
@ealonso LGTM! :sparkles:

**Comment by ealonso** [comment]
ci:forward

File: modules/apps/asset/source-formatter-suppressions.xml [hunk 1]
```diff
@@ -6,6 +6,7 @@
 		<suppress checks="PersistenceCallCheck" files="asset-service/src/main/java/com/liferay/asset/internal/search/AssetSearcher\.java" />
 	</checkstyle>
 	<source-check>
+		<suppress checks="JavaTestMethodAnnotationsCheck" files="asset-publisher-test/src/testIntegration/java/com/liferay/asset/publisher/frontend/taglib/form/navigator/test/AssetPublisherConfigurationFormNavigatorTest.java" />
```
Comment (dsanz) [inline modules/apps/asset/source-formatter-suppressions.xml:9]: BChan might dislike this

**Comment by dsanz** [comment]
LGTM ;)

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by ealonso** [comment]
Manual forward see: https://github.com/brianchandotcom/liferay-portal/pull/150611

**Pull request closed without merging**